### PR TITLE
Migrate test_peak_mem_multi_tensor_optimizers to OptimizerInfo

### PR DESCRIPTION
**Pull request by janeyx99**
Replace the following:
```
(pytorch-3.10) [janeyx@devgpu023.odn1 ~/local/pytorch (1bbf1c6f)]$ python test/test_optim.py -k test_peak_mem_multi_tensor_optimizers
/home/janeyx/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.17.3 and <1.25.0 is required for this version of SciPy (detected version 1.26.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
.
----------------------------------------------------------------------
Ran 1 test in 38.599s

OK
```

with 11 tests (one for each foreach optim :))
```
(pytorch-3.10) [janeyx@devgpu023.odn1 ~/local/pytorch (1bbf1c6f)]$ python test/test_optim.py -k TestOptimRenewedCUDA.test_foreach_memory
/home/janeyx/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.17.3 and <1.25.0 is required for this version of SciPy (detected version 1.26.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
...........
----------------------------------------------------------------------
Ran 11 tests in 39.293s

OK
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115299
* #115025
* __->__ #115023
* #114802

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -140,6 +140,63 @@ def test_set_default_dtype_works_with_foreach(self, device, dtype, optim_info):
...
 
+    @onlyCUDA
+    @optims([optim for optim in optim_db if "foreach" in optim.supported_impls], dtypes=[torch.float32])
+    def test_peak_memory_foreach(self, device, dtype, optim_info):
+        nparams = 10
+        optim_inputs = optim_info.optim_inputs_func()
+        optim_cls = optim_info.optim_cls
+        for optim_input in optim_inputs:
+            kwargs = deepcopy(optim_input.kwargs)
+            if kwargs.get("capturable", False) and optim_cls.__name__ == "ASGD":
```
Comment (albanD) [inline test/test_optim.py:151]: nit: do we want to encore skips in the test here or as a skip on the OptimInfo?

**Review by albanD — APPROVED** [review]
SGTM!

File: test/test_optim.py (hunk 1 above)
Comment (janeyx99) [reply test/test_optim.py:152]: This is a good point, I'll look into that. I see this as an unfortunate skip vs a necessary skip. Are DecorateInfos used mostly for permanent skips or temporary ones?

Comment (albanD) [reply test/test_optim.py:152]: I think one of the feedback we had from OpInfo is that it is confusing to have skips in different places and they should be centralized.
I think we should strive to get there for this new version as well.

Comment (janeyx99) [reply test/test_optim.py:152]: Okay, I'll put everything in DecorateInfo for now

**Pull request closed without merging**